zookeeper/core/component.py

```diff
@@ -289,14 +291,20 @@ def _list_field_strings(instance, color: bool, single_line: bool) -> Iterator[st
         except AttributeError as e:
             if field.allow_missing:
                 value = utils.missing
+                yield f"{field_name}={value}" if color else f"{field_name}={value}"
+                continue
```

Comment:
Could you explain what the bug is here?

Reply:
A few lines further down, you call `base_getattr` on this missing value, which throws an error.

Reply:
Ah of course, nice find :+1:

Do you mind if we instead change `if parent_instance is not None:` to `if value is not utils.missing and parent_instance is not None:`?

Reply:
Yeah I somewhat do, if the value is `utils.missing` anyway, why would we even care to check what the parent instance is? It can't be inherited (or maybe it can, but it's meaningless). On the other hand it may make the code more maintainable if we don't have to duplicate that yield value, so it might not be too bad

Reply:
Ah but that's exactly what my change would do - skip the inheritence/parent instance check if it's a missing value. It'll just mean that we don't have to repeat the `yield` statement which I think is cleaner.

Reply:
We'd still run the following code:
```python
parent_instance = next(
            utils.generate_component_ancestors_with_field(instance, field_name), None
        )

if utils.is_component_instance(value):
            if is_inherited:
                value = "<inherited component instance>"
            elif single_line:
                value = repr(value)
            else:
                value = f"\n{INDENT}".join(str(value).split("\n"))
        elif is_inherited:
            value = "<inherited value>"
        elif callable(value):
            value = "<callable>"
        elif isinstance(value, str):
            value = f'"{value}"'
```
All of which is unnecessary. But sure, I'll do that anyway